login-service/src/api-login/dto/user-inputs.dto.ts

```diff
@@ -0,0 +1,123 @@
+import { HttpException, HttpStatus } from "@nestjs/common";
+import { RegistrationTokenInput } from "./link-user.dto";
+
+/**
+ * Base input type containing all needed fields to create a new user
+ */
+export class BaseUserInput {
+    /**
+     * The username to set for the new user.
+     * Must be given and cannot be empty
+     *
+     * @example "testUser"
+     */
+    username: string;
+
+    /**
+     * The name to display in the UI for the new user.
+     * Must be given and can't be empty
+     *
+     * @example "Test User"
+     */
+    displayName: string;
+
+    /**
+     * The email of the new user.
+     * Can be ommitted, but if given can't be empty.
+     *
+     * @example "test-user@example.com"
+     */
+    email?: string;
+
+    /**
+     * Checks for a valid {@link BaseUserInput}
+     *
+     * Needed:
+     * - `username` is a required, non empty string
+     * - `displayName` is a required, non empty string
+     * - `email` is, if given, a non empty string
+     *
+     * @param input The instance to check
+     * @returns The argument unchanged
+     * @throws {@link HttpException} BAD_REQUEST if invalid
+     */
+    static check(input: BaseUserInput): BaseUserInput {
+        if (!input.username || input.username.trim().length <= 0) {
```

Comment:
use `== 0` for unsigned comparison

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#description
it is a [precise integers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) with the number of utf16 units. 0 is explicitly given and no string can be shorter than 0 utf16 units

Reply:
Didn't I fix this already?